freqtrade/optimize/backtesting.py

```diff
@@ -63,9 +63,7 @@ def __init__(self, config: Dict[str, Any]) -> None:
         self.all_results: Dict[str, Dict] = {}
 
         self.exchange = ExchangeResolver.load_exchange(self.config['exchange']['name'], self.config)
-
-        dataprovider = DataProvider(self.config, self.exchange)
-        IStrategy.dp = dataprovider
+        self.dataprovider = DataProvider(self.config, self.exchange)
```

Comment:
I guess "this" dataprovider should be called without exchange.

This wasn't possible before (mypy :eyes:)- but if you need ANYTHING from the dataprovider that requires an exchange class during backtesting, you're doing something wrong - as you're using "now" data to analyze historic data.